jni/src/common/Foreign/JNI/Unsafe.hs

```diff
@@ -20,6 +26,74 @@
 -- Reexports definitions from "Foreign.JNI.Unsafe.Internal".
 module Foreign.JNI.Unsafe
   ( module Foreign.JNI.Unsafe.Internal
+  , getMethodID
+  , getStaticMethodID
   ) where
 
-import Foreign.JNI.Unsafe.Internal
+import Control.Exception (Exception, catch, throwIO)
+import Data.List (intercalate)
+import Data.Singletons (sing)
+import qualified Data.Text as Text
+import qualified Data.Text.Encoding as Text
+import Foreign.JNI.Types
+import qualified Foreign.JNI.String as JNI
+import qualified Foreign.JNI.Unsafe.Internal as Internal
+import Foreign.JNI.Unsafe.Internal hiding (getMethodID, getStaticMethodID)
+import Foreign.JNI.Unsafe.Internal.Introspection
+
+data NoSuchMethod = NoSuchMethod
+  { className :: Text.Text
+  , methodName :: Text.Text
+  , targetSignature :: String
```

Comment:
Because it was generated directly from `show` ; it would have generated an unnecessary `unpack . pack`.
Now that `Text` is used internally in `instance Show NoSuchMethod`, this back and forth is necessary, the signature was updated.